### PR TITLE
XU[34]: Add new mali packages for gbm/kms support

### PR DESCRIPTION
**Pull request by Ntemis**

### CODE REVIEW

File: projects/OdroidXU3/bootloader/boot.ini [hunk 1]
```diff
@@ -12,23 +12,9 @@ setenv macaddr "00:1e:06:61:7a:39"
...
-setenv console "console=ttySAC2,115200n8"
-
-#------------------------------------------------------------------------------------------------------
-#
-# OpenELEC variables
-#
-# Setup the OpenELEC options
-#   valid values are: progress ssh debugging quiet break verbose
-#------------------------------------------------------------------------------------------------------
-setenv openelec "quiet consoleblank=0 loglevel=2 vt.global_cursor_default=0"
```
Comment (kivutar) [inline projects/OdroidXU3/bootloader/boot.ini]: Note: `consoleblank=0 loglevel=2 vt.global_cursor_default=0` is needed

File: projects/OdroidXU3/bootloader/boot.ini [hunk 2]
```diff
@@ -237,16 +225,16 @@ setenv hdmi_clk_amp_lvl     "31"
...
 #------------------------------------------------------------------------------
 
-setenv hdmi_phy_control "hdmi_tx_amp_lvl=${hdmi_tx_amp_lvl} hdmi_tx_lvl_ch0=${hdmi_tx_lvl_ch0} hdmi_tx_lvl_ch1=${hdmi_tx_lvl_ch1} hdmi_tx_lvl_ch2=${hdmi_tx_lvl_ch2} hdmi_tx_emp_lvl=${hdmi_tx_emp_lvl} hdmi_clk_amp_lvl=${hdmi_clk_amp_lvl} hdmi_tx_res=${hdmi_tx_res} HPD=${HPD} vout=${vout}"
+# setenv hdmi_phy_control "hdmi_tx_amp_lvl=${hdmi_tx_amp_lvl} hdmi_tx_lvl_ch0=${hdmi_tx_lvl_ch0} hdmi_tx_lvl_ch1=${hdmi_tx_lvl_ch1} hdmi_tx_lvl_ch2=${hdmi_tx_lvl_ch2} hdmi_tx_emp_lvl=${hdmi_tx_emp_lvl} hdmi_clk_amp_lvl=${hdmi_clk_amp_lvl} hdmi_tx_res=${hdmi_tx_res} HPD=${HPD} vout=${vout}"
 
 
 # Load kernel, initrd and dtb in that sequence
 fatload mmc 0:1 0x40008000 KERNEL
-fatload mmc 0:1 0x44000000 exynos5422-odroidxu3.dtb
+fatload mmc 0:1 0x44000000 exynos5422-odroidxu4.dtb
```
Comment (kivutar) [inline projects/OdroidXU3/bootloader/boot.ini]: The xu4 boots fine with the xu3 dtb, I'm not sure about the opposite.

File: projects/OdroidXU3/packages/Mali_OpenGL_ES_SDK/package.mk [hunk 3]
```diff
@@ -41,5 +41,9 @@ make_target() {
 
 makeinstall_target() {
   mkdir -p $SYSROOT_PREFIX/usr/include
-    cp -PR usr/include/* $SYSROOT_PREFIX/usr/include
+    cp -PR $PKG_BUILD/inc/* $SYSROOT_PREFIX/usr/include
+    cp -PR $PKG_BUILD/simple_framework/inc/mali/* $SYSROOT_PREFIX/usr/include
+
+    mkdir -p $SYSROOT_PREFIX/usr/lib/pkgconfig
+    cp -PRv $PKG_DIR/pkgconfig/*.pc $SYSROOT_PREFIX/usr/lib/pkgconfig
```
Comment (kivutar) [inline projects/OdroidXU3/packages/Mali_OpenGL_ES_SDK/package.mk]: All these pkgconfig files are not needed, I was able to build and link RA without them

File: projects/OdroidXU3/packages/odroidxu3-mali/package.mk [hunk 4]
```diff
@@ -19,48 +19,47 @@
...
 
 PKG_IS_ADDON="no"
 PKG_AUTORECONF="no"
+LIBMALI_ARCH="arm-linux-gnueabihf"
+LIBMALI_FILE="libmali.so"
 
-#unpack() {
-#  $SCRIPTS/extract $PKG_NAME $(basename $PKG_URL) $BUILD
-#}
+configure_target() {
```
Comment (kivutar) [inline projects/OdroidXU3/packages/odroidxu3-mali/package.mk]: Is this callback needed? I can build without it in my PR

File: projects/OdroidXU3/bootloader/boot.ini (hunk 2 above)
Comment (Ntemis) [reply projects/OdroidXU3/bootloader/boot.ini]: Can you please test me this commit
https://github.com/Ntemis/Lakka-LibreELEC/commit/2af2aac7c4709e1d7cded4d36c04eb52f2cb2f48
Thanks

Comment (kivutar) [reply projects/OdroidXU3/bootloader/boot.ini]: Sure, but I'm getting a

    /home/kivutar/Lakka-LibreELEC/build.Lakka-OdroidXU3.arm-2.1-devel/toolchain/lib/gcc/armv7ve-libreelec-linux-gnueabi/6.2.0/../../../../armv7ve-libreelec-linux-gnueabi/bin/ld.gold: error: cannot find -lGLESv1_CM
collect2: error: ld returned 1 exit status

When linking RA

File: projects/OdroidXU3/bootloader/boot.ini (hunk 1 above)
Comment (Ntemis) [reply projects/OdroidXU3/bootloader/boot.ini]: can you please test this commit:
https://github.com/Ntemis/Lakka-LibreELEC/commit/ba5047292d01dc1e5a7df5734db29692aa3f6b2f

Comment (kivutar) [reply projects/OdroidXU3/bootloader/boot.ini]: Sure, but I'm getting a

    /home/kivutar/Lakka-LibreELEC/build.Lakka-OdroidXU3.arm-2.1-devel/toolchain/lib/gcc/armv7ve-libreelec-linux-gnueabi/6.2.0/../../../../armv7ve-libreelec-linux-gnueabi/bin/ld.gold: error: cannot find -lGLESv1_CM
    collect2: error: ld returned 1 exit status

When linking RA

**Comment by Ntemis** [comment]
XU4 boots and retroarch works flawless.
Now all you need to do it test on XU3 :)

**Comment by kivutar** [comment]
Awesome news @Ntemis ! I will try it tonight.

One last thing, can you try on your side to add my gbm patch and remove mesa as dependancy?

If not, I will try it on my side.

**Pull request merged**